demo/apps/todomvc/basis/list/list.js

```diff
@@ -0,0 +1,48 @@
+var Node = require('basis.ui').Node;
+var Value = require('basis.data').Value;
+var Expression = require('basis.data.value').Expression;
+var setAccumulateState = require('basis.data').Dataset.setAccumulateState;
+var Task = require('../task');
+var Item = require('../item/item');
+var route = require('../route');
+
+module.exports = new Node({
+  template: resource('./list.tmpl'),
+  dataSource: route.as(function(mode){
+    if (mode === 'active') {
+      return Task.active;
+    }
+    else if (mode === 'completed') {
+      return Task.completed;
+    }
+
+    return Task.all;
+  }),
+  sorting: 'data.created',
+  childClass: Item,
+  action: {
+    toggle: function(){
+      setAccumulateState(true);
```

Comment:
Why that is needed?

Reply:
A little optimization, cause we do `item.set('completed', false);` for every item